source/MaterialXCore/Document.h

```diff
@@ -98,6 +121,33 @@ class MX_CORE_API Document : public GraphElement
     /// nodes, and include both Input and Output elements.
     vector<PortElementPtr> getMatchingPorts(const string& nodeName) const;
 
+    /// Return the child element, if any, with the given document and data library
+    template <class T> shared_ptr<T> getChildOfType(const string& name) const
```

Comment:
@ashwinbhat I've had a chance to implement the move of `getChildOfType` and `getChildrenOfType` from Element.h to `Element.cpp`, which would allow these two non-virtual overloads to be removed:

https://github.com/AcademySoftwareFoundation/MaterialX/pull/2069

Although non-virtual overloaded methods are allowed in C++, we strongly avoid this pattern in MaterialX because of the hazards for unwary developers, and static analysis packages like PVS-Studio emit warnings when non-virtual overloads are detected in the code!

What are your thoughts?  If I went ahead and merged my change linked above, would you be open to updating your change to take advantage of it?

Reply:
@jstone-lucasfilm this is great. Please go ahead and merge, I'll update my PR to take advantage of this.

Reply:
@ashwinbhat Done!